azext_edge/edge/providers/orchestration/resources/dataflows.py

```diff
@@ -21,38 +24,152 @@
-        )
+        self.instances = Instances(cmd=cmd)
+        self.iotops_mgmt_client = self.instances.iotops_mgmt_client
         self.ops: "DataflowProfileOperations" = self.iotops_mgmt_client.dataflow_profile
         self.dataflows = DataFlows(self.iotops_mgmt_client.dataflow)
 
+    def create(
+        self,
+        name: str,
+        instance_name: str,
+        resource_group_name: str,
+        profile_instances: Optional[int] = 1,
+        log_level: Optional[str] = "info",
+        **kwargs,
+    ):
+
+        resource = {
+            "properties": {
+                "diagnostics": {
+                    "logs": {"level": log_level},
+                },
+                "instanceCount": profile_instances,
+            },
+            "extendedLocation": self.instances.get_ext_loc(
+                name=instance_name, resource_group_name=resource_group_name
+            ),
+        }
+
+        with console.status(f"Creating {name}..."):
+            poller = self.ops.begin_create_or_update(
+                instance_name=instance_name,
+                dataflow_profile_name=name,
+                resource_group_name=resource_group_name,
+                resource=resource,
+            )
+            return wait_for_terminal_state(poller, **kwargs)
+
+    def update(
+        self,
+        name: str,
+        instance_name: str,
+        resource_group_name: str,
+        profile_instances: Optional[int] = None,
+        log_level: Optional[str] = None,
+        **kwargs,
+    ):
+        # get the existing dataflow profile
+        original_profile = self.show(
+            name=name,
+            instance_name=instance_name,
+            resource_group_name=resource_group_name,
+        )
+
+        # update the properties
+        if profile_instances:
+            original_profile["properties"]["instanceCount"] = profile_instances
+        if log_level:
+            original_profile["properties"]["diagnostics"]["logs"]["level"] = log_level
+
+        with console.status(f"Updating {name}..."):
+            poller = self.ops.begin_create_or_update(
+                instance_name=instance_name,
+                dataflow_profile_name=name,
+                resource_group_name=resource_group_name,
+                resource=original_profile,
+            )
+            return wait_for_terminal_state(poller, **kwargs)
+
+    def delete(
+        self,
+        name: str,
+        instance_name: str,
+        resource_group_name: str,
+        confirm_yes: Optional[bool] = None,
+        **kwargs,
+    ) -> dict:
+        dataflows = self.dataflows.list(
+            dataflow_profile_name=name,
+            instance_name=instance_name,
+            resource_group_name=resource_group_name,
+        )
+        dataflows = list(dataflows)
+
+        if name == "default":
+            logger.warning(
+                "Deleting the 'default' dataflow profile may cause disruptions."
```

Comment:
nit: Deleting the 'default' dataflow profile may cause disruptions for configured dataflow(?)

Reply:
def more than just a nit

Reply:
it will also affect quick start so I'm going to leave it very general